File: software/zyarm_sdk/cpp/examples/teleop_step.cpp
#include <iostream>
#include <memory>

#include "zyarm_sdk/teleop.hpp"

int main()
{
  zyarm_sdk::ZyArmConfig leader_config;
  leader_config.port = "/dev/ttyUSB0";
  zyarm_sdk::ZyArmConfig follower_config;
  follower_config.port = "/dev/ttyUSB1";

  auto leader = std::make_shared<zyarm_sdk::ZyArm>(leader_config);
  auto follower = std::make_shared<zyarm_sdk::ZyArm>(follower_config);
  zyarm_sdk::ZyArmTeleopPair pair(leader, follower);
  pair.connect();
  pair.start_step_mode();
  auto result = pair.step(true, false);
  if (result.has_value()) {
    std::cout << "teleop action gripper=" << result->action.positions[6] << "\n";
  }
  pair.close();
  return 0;
}

File: software/zyarm_sdk/python/examples/teleop_step.py
from zyarm_sdk import ZyArm, ZyArmConfig
from zyarm_sdk.teleop import ZyArmTeleopPair


def main() -> None:
    leader = ZyArm(ZyArmConfig(port="/dev/ttyUSB0"))
    follower = ZyArm(ZyArmConfig(port="/dev/ttyUSB1"))
    pair = ZyArmTeleopPair(leader, follower).connect()
    try:
        pair.start_step_mode()
        result = pair.step(wait_action=True)
        if result is not None:
            print(result.action.positions)
    finally:
        pair.close()


if __name__ == "__main__":
    main()
